EasyPost/Exceptions/API/_Base.cs

```diff
@@ -75,7 +76,16 @@ internal static ApiError FromErrorResponse(RestResponse response)
                 // try to extract error details from the API response
                 Dictionary<string, Dictionary<string, object>> body = JsonSerialization.ConvertJsonToObject<Dictionary<string, Dictionary<string, object>>>(response.Content);
 
-                errorMessage = body["error"]["message"].ToString();
+                // Errors from some carriers may be an array improperly assigned to the `message` field, check for those here (eg: GSO)
+                // TODO: This is a bandage fix until the GSO service maps its error messages correctly. More details can be found here:
+                // https://github.com/EasyPost/easypost-php/pull/189
```

Comment:
I apologize, this comment is no longer true as written. Something like this should do:

```suggestion
                // Errors may be an array improperly assigned to the `message` field instead of the `errors` field, concatenate those here
```

Reply:
Resolved